Create map to save & show ubication

angel23v opened the pull request:
## Quick Info
   - Map to save the ubication where you click
   - Search button tool
   - Show map with the ubication saved when you edit an event
   - Navigate inside the map with the cursor
   - FIll automatically ubication fields on the form  (ubicacion, latitud, longitud)

Migrations? 
Yes, added latitud & longitud columns

## What does this change?
The way to save & show the ubication

## Why are you changing that?
Because it looks better with a interactive map and makes a better experience for the user 

## How do you manually test this?
Clicking many times on the map to see if the infomation was showed on the ubication fields.
Saving events and see the correct ubication info on my view
Editing events to see if the ubicatiion saved was showed on the map

## Screenshots
![add_event](https://github.com/BrightCoders-Institute/s5a1-administrador-de-eventos-team4/assets/86389712/918900fe-797c-4245-92a3-2241c6a60239)
![show_and_edit_ubi](https://github.com/BrightCoders-Institute/s5a1-administrador-de-eventos-team4/assets/86389712/dbf1fa08-ba9b-43f0-8cff-0ae7f268d96d)

angel23v committed:
Filter view on events

When you create a public event, all the users will be able to see it but if you create a private event you just gonna see it

angel23v committed:
Missing letter

angel23v committed:
Creating a interactive map to save ubication

Using leaflet gem and a free API to convert the coordinates to the information for the place

angel23v committed:
Show ubication saved correctly

When u want to edit an event you will see the ubcation saved on the map

MauricioNC commented on app/javascript/controllers/map_controller.js:
```diff
@@ -0,0 +1,65 @@
+    let longitud = document.querySelector(".longitud");
+    let ubicacion = document.querySelector(".ubicacion");
+
+    const apiKey = "pk.8ef046976867e15790d27dae0c26d553";
```
Don't leave this api key directly in the code, please move it to an environment variable instead

angel23v replied:
is there any documentation?

MauricioNC replied:
Personally, I've never faced to this kind of situation. May be a solution could be save that api key into the credentials file and then create a method that returns the api key. 

To edit this file you can run `rails credentials:edit`.

And just call this method from Stimulus.

angel23v replied:
Thanks I try with that option & I wasn`t able to solve but searching I found a gem call "dotenv" and it works for me.

MauricioNC commented on app/views/layouts/_navbar.html.erb:
```diff
@@ -2,12 +2,16 @@
-    </button>
     <div class="collapse navbar-collapse" id="navbarSupportedContent">
       <ul class="navbar-nav me-auto mb-2 mb-lg-0">
+          <% if !signed_in?%>
+            <p class= "text-light">SISTEMA DE GESTION DE EVENTOS</p> 
+          <% end %>
+        <li class="nav-item">
+          <% if signed_in?%>
+            <%= link_to "Sistema de Gestion Eventos", events_main_path , class: "btn btn-dark text-light"%>
+          <% end %>
+        </li>
         <li class="nav-item">
```
You can improve the logic of this code. Have you used the unless statement?

angel23v replied:
Yes, It will be better if I use ternary operator?

MauricioNC replied:
No, ternary operator is recommended to use for short evaluations.

- In the first if, you are evaluating a negative condition, you can use unless instead.
- In the second if, you have the li tag outside of the if statement, if the condition is not met, you will have an empty li tag, so you can move the li tag inside the if statement, by this way, if the condition is not met, the li tag will not be created.

MauricioNC replied:
or you can use an if else statement, but sometimes the linter marks a suggestion to replace the negative conditions to unless statement.

angel23v replied:
Thanks I already try with unless statement

MauricioNC commented on app/views/layouts/application.html.erb:
```diff
@@ -5,14 +5,21 @@
     <%= csrf_meta_tags %>
     <%= csp_meta_tag %>
-
+     
```
Remove blank spaces

angel23v committed:
Attend issues 2 of 3

angel23v committed:
Save API in a environment

Using gem dotenv, load the gem in the application.rb & create a controller for get the gem which is in a file call ".env"  and in the JS with a fetch request get the API

MauricioNC reviewed (approved):
LGTM!

juperameza reviewed (approved):
LGTM
